lib/libvarnish/vss.c

```diff
@@ -100,6 +101,42 @@ vss_parse(char *str, char **addr, char **port)
 	return (NULL);
 }
 
+static const char *
+vss_parse_range(char *str, char **addr, char **port, unsigned long *lo,
+    unsigned long *hi)
+{
+	const char *errp;
+	char *end;
+	unsigned long l, h;
+
+	errp = vss_parse(str, addr, port);
+	if (errp != NULL)
+		return (errp);
+	if (*port == NULL || **port == '-')
+		return (NULL);
+
+	l = strtoul(*port, &end, 10);
+	if (end[0] != '-' || end[1] == '\0')
+		return (NULL);
+	if (strchr(end + 1, '-') != NULL)
```

Comment:
why do we special case a second hyphen check here? Should the `end[0] != '\0'` not suffice?

Reply:
That is because `strtoul()` "helpfully" interprets `-81` as a negative number, so a port range defined as `80--81` would result in a `Range end higher than 65535` error message. This line should guard against any attempt at making a negative numbers, even if there is whitespace inbetween (`strtoul()` also ignores leading whitespace).

Reply:
that is a very good reason, thank you!